### PR TITLE
Restructure and expand the build lifecycle regarding dependencies, plugins and test coverage

### PR DESCRIPTION
**Pull request by d0gb0t666**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,128 +1,196 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (britter) [inline pom.xml:2]: This file has changed from two spaces indent to four spaces indent, which makes it impossible to review the actual changes. If the code style should change, I'd recommend doing this in a separate PR.

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+logging:
+  level:
+    root: info
+spring:
+  application:
+    name: deckcompare
+  main:
+    banner-mode: console
```
Comment (britter) [inline src/main/resources/application.yml:8]: console is the default value. See https://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+server:
+  port: 9090
```
Comment (britter) [inline src/main/resources/application.yml:10]: Why?

Comment (britter) [inline src/main/resources/application.yml:10]: Missing new line at EOF

File: src/test/java/de/flux/playground/deckcompare/DeckcompareIT.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(Deckcompare.class)
+public class DeckcompareIT {
+
+	@Test
+	public void contextLoads() {
```
Comment (britter) [inline src/test/java/de/flux/playground/deckcompare/DeckcompareIT.java:13]: Empty Test?

File: pom.xml [hunk 5]
```diff
@@ -1,128 +1,196 @@
...
+    <description>Deck comparator for Android: Netrunner decks.</description>
+    <url>http://flux.de/deckcompare</url>
+
+    <properties>
+        <java.version>1.8</java.version>
+        <maven.compiler.source>${java.version}</maven.compiler.source>
+        <maven.compiler.target>${java.version}</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <project.dependency.spring-boot.version>1.3.3.RELEASE</project.dependency.spring-boot.version>
```
Comment (britter) [inline pom.xml:23]: What do we gain by moving this into the properties section? We only need this once.

File: pom.xml [hunk 6]
```diff
@@ -1,128 +1,196 @@
...
+        <project.plugin.spring-boot-maven-plugin.version>${project.dependency.spring-boot.version}</project.plugin.spring-boot-maven-plugin.version>
```
Comment (britter) [inline pom.xml:24]: This is not needed when the build declares

```
  <parent>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-parent</artifactId>
    <version>1.3.3.RELEASE</version>
  </parent>
```

This way the correct version of the plugin will be inherited. Furthermore the spring-boot-dependencies block can also be dropped.

File: pom.xml [hunk 7]
```diff
@@ -1,128 +1,196 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.16.8</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <pluginManagement>
```
Comment (britter) [inline pom.xml:67]: Why do we need `pluginManagement`? We only have this nodule at the moment. Using plugin management at this point complicates things.

File: pom.xml (hunk 1 above)
Comment (d0gb0t666) [inline pom.xml:2]: Next time if I knew better about the context of a PR...

File: src/main/resources/application.yml (hunk 2 above)
Comment (d0gb0t666) [inline src/main/resources/application.yml:8]: Yep, but the entriy shows how to configure the banner-mode.

File: src/main/resources/application.yml (hunk 3 above)
Comment (d0gb0t666) [inline src/main/resources/application.yml:10]: To show what can be configured in the configuration file.

File: src/test/java/de/flux/playground/deckcompare/DeckcompareIT.java (hunk 4 above)
Comment (d0gb0t666) [inline src/test/java/de/flux/playground/deckcompare/DeckcompareIT.java:13]: Yep, this way you can test if the context loads completely without errors.

File: src/main/resources/application.yml (hunk 2 above)
Comment (britter) [inline src/main/resources/application.yml:8]: Sorry, didn't get that from the PR title :-)

File: pom.xml (hunk 5 above)
Comment (d0gb0t666) [inline pom.xml:23]: Actually we need the spring related version two times. One time for the dependency and the second time for the plugin. To handle this through properties allows us to easily switch to another version and to handle different versions for dependency and plugin if necessary.

File: pom.xml (hunk 6 above)
Comment (d0gb0t666) [inline pom.xml:24]: Correct, but it seems that Jan don't want a parent. To add the parent-pom to the dependencyManagement is in either way not the best solution. Again, this only shows possibilities...

File: pom.xml (hunk 7 above)
Comment (d0gb0t666) [inline pom.xml:67]: We don't need it. But it has become naturalized to configure each plugin in the management and only declare the usage in the build section.

File: pom.xml [hunk 8]
```diff
@@ -7,7 +7,7 @@
 
     <groupId>de.flux.playground</groupId>
     <artifactId>deckcompare</artifactId>
-    <version>0.0.1-SNAPSHOT</version>
+    <version>1.0.0-SNAPSHOT</version>
```
Comment (JLengenfeld) [inline pom.xml:10]: Why did you change the version? The project right now is not even in a real functional stage... XD
